### PR TITLE
Break down the lnd_test.go file into smaller files

### PR DESCRIPTION
**Pull request by yyforyongyu**
This PR aims at breaking down the large file `lnd_test.go` into smaller files for better maintenance. Tests are grouped based on loosely defined categories. While it's far from perfect, this should set sail for refactoring the `itest` to get rid of test flakes.

All the changes are just moving code into different files, no lines are changed, except these two commits,
- `itest: use require in assertions`
- `itest: refactor assertNumConnections to use wait.NoError`

### CODE REVIEW

**Comment by yyforyongyu** [comment]
Wonder how Coveralls count the lines as this commit has decreased lines of code, but it still says the coverage decreased🧐

File: lntest/itest/assertions.go [hunk 1]
```diff
@@ -277,51 +276,44 @@ func assertChannelClosed(ctx context.Context, t *harnessTest,
...
 		}
 
 		// Assert that the channel is in coop broadcasted.
-		if !strings.Contains(channel.ChanStatusFlags,
-			channeldb.ChanStatusCoopBroadcasted.String()) {
-			t.Fatalf("channel not coop broadcasted, "+
-				"got: %v", channel.ChanStatusFlags)
-		}
+		require.Contains(
+			t.t, channeldb.ChanStatusCoopBroadcasted.String(),
```
Comment (guggero) [inline lntest/itest/assertions.go:297]: I think the arguments need to be swapped here.

File: lntest/itest/assertions.go [hunk 2]
```diff
@@ -615,9 +604,7 @@ func shutdownAndAssert(net *lntest.NetworkHarness, t *harnessTest,
 	err := wait.NoError(func() error {
 		return net.ShutdownNode(node)
 	}, defaultTimeout)
-	if err != nil {
-		t.Fatalf("unable to shutdown %v: %v", node.Name(), err)
-	}
+	require.NoError(t.t, err, "unable to shutdown %v", node.Name())
```
Comment (guggero) [inline lntest/itest/assertions.go:600]: Needs to be `NoErrorf` for the `%v` to be interpreted.

File: lntest/itest/assertions.go [hunk 3]
```diff
@@ -1378,15 +1336,12 @@ func assertDLPExecuted(net *lntest.NetworkHarness, t *harnessTest,
...
-	}
+	require.NoError(t.t, err, "unable to get dave's balance")
 
 	daveBalance := daveBalResp.ConfirmedBalance
-	if daveBalance <= daveStartingBalance {
-		t.Fatalf("expected dave to have balance above %d, "+
-			"instead had %v", daveStartingBalance, daveBalance)
-	}
+	require.Greater(
+		t.t, daveBalance, daveStartingBalance, "balance no increased",
```
Comment (guggero) [inline lntest/itest/assertions.go:1336]: nit: s/no/not/

File: lntest/itest/assertions.go [hunk 4]
```diff
@@ -1744,23 +1687,20 @@ func sendAndAssertFailure(t *harnessTest, node *lntest.HarnessNode,
...
-	if err != nil {
-		t.Fatalf("unable to get payment result: %v", err)
-	}
+	require.NoError(t.t, err, "unable to get payment result")
 
-	if result.Status != lnrpc.Payment_FAILED {
-		t.Fatalf("payment was expected to fail, but succeeded")
-	}
+	require.Equal(
+		t.t, result.Status, lnrpc.Payment_FAILED,
```
Comment (guggero) [inline lntest/itest/assertions.go:1689]: nit: expected result needs to come first for proper error message. Same in the `Equal` case below.

**Review by guggero — COMMENTED** [review]
This is great! Wanted to do this for a while but never got to it.

There are a few small things that should be fixed, but otherwise LGTM :100:

**Comment by guggero** [comment]
> Wonder how Coveralls count the lines as this commit has decreased lines of code, but it still says the coverage decreasedmonocle_face

Haha, I wonder how that tool works sometimes too. Though I stopped looking at its results in the PRs a while ago since it seems so random...

File: lntest/itest/assertions.go (hunk 1 above)
Comment (yyforyongyu) [reply lntest/itest/assertions.go:297]: Fixed!

File: lntest/itest/assertions.go (hunk 2 above)
Comment (yyforyongyu) [reply lntest/itest/assertions.go:600]: Fixed!

File: lntest/itest/assertions.go (hunk 3 above)
Comment (yyforyongyu) [reply lntest/itest/assertions.go:1336]: Fixed!

File: lntest/itest/assertions.go (hunk 4 above)
Comment (yyforyongyu) [reply lntest/itest/assertions.go:1689]: Fixed!

**Review by yyforyongyu — COMMENTED** [review]
> > Wonder how Coveralls count the lines as this commit has decreased lines of code, but it still says the coverage decreasedmonocle_face
> 
> Haha, I wonder how that tool works sometimes too. Though I stopped looking at its results in the PRs a while ago since it seems so random...

Haha yeah I really want to get an all green build once in a lifetime🤓

**Comment by yyforyongyu** [comment]
Bump this one.

**Review by guggero — APPROVED** [review]
Great refactor, LGTM :100:

**Comment by guggero** [comment]
Itest fails because of new errors that need to be white listed:

```
<time> [ERR] CNCT: ChannelArbitrator(<chan_point>): unable to mark commitment broadcasted: channel not found
<time> [ERR] RPCS: [/lnrpc.Lightning/CloseChannel]: rpc error: code = Canceled desc = context canceled
```

**Comment by yyforyongyu** [comment]
> Itest fails because of new errors that need to be white listed:
> 
> ```
> <time> [ERR] CNCT: ChannelArbitrator(<chan_point>): unable to mark commitment broadcasted: channel not found
> <time> [ERR] RPCS: [/lnrpc.Lightning/CloseChannel]: rpc error: code = Canceled desc = context canceled
> ```

Rebased!

**Comment by yyforyongyu** [comment]
`itest` got only one failure, so close!!!

**Comment by guggero** [comment]
Wow, nice! We're so close I can almost smell the green :D

File: lntest/itest/assertions.go [hunk 5]
```diff
@@ -543,56 +543,49 @@ func assertNumOpenChannelsPending(ctxt context.Context, t *harnessTest,
...
 
 // assertNumConnections asserts number current connections between two peers.
-// TODO(yy): refactor to use wait.
 func assertNumConnections(t *harnessTest, alice, bob *lntest.HarnessNode,
 	expected int) {
 	ctxb := context.Background()
+	ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
 
-	const nPolls = 10
+	err := wait.NoError(func() error {
```
Comment (Roasbeef) [inline lntest/itest/assertions.go:551]: 👍

**Review by Roasbeef — APPROVED** [review]
LGTM 💈

**Comment by Roasbeef** [comment]
Gonna cause a ton of conflicts w/ open PRs, but I think it's worth it as it makes the itests much more manageable, which may aide in flake hunting.

**Pull request merged**